Package.swift

```diff
@@ -47,54 +48,65 @@ let isLinux = false
-    .target(name: "SwiftGRPC",
-            dependencies: ["CgRPC", "SwiftProtobuf"]),
-    .target(name: "SwiftGRPCNIO",
-            dependencies: [
-              "NIOFoundationCompat",
-              "NIOHTTP1",
-              "NIOHTTP2",
-              "SwiftProtobuf"]),
-    .target(name: "CgRPC",
-            dependencies: cGRPCDependencies),
-    .target(name: "RootsEncoder"),
-    .target(name: "protoc-gen-swiftgrpc",
-            dependencies: [
-              "SwiftProtobuf",
-              "SwiftProtobufPluginLibrary",
-              "protoc-gen-swift"]),
-    .target(name: "BoringSSL"),
-    .target(name: "Echo",
-            dependencies: [
-              "SwiftGRPC",
-              "SwiftProtobuf",
-              "Commander"],
-            path: "Sources/Examples/Echo"),
-    .target(name: "EchoNIO",
-            dependencies: [
-              "SwiftGRPCNIO",
-              "SwiftProtobuf",
-              "Commander"],
-            path: "Sources/Examples/EchoNIO"),
-    .target(name: "Simple",
-            dependencies: ["SwiftGRPC", "Commander"],
-            path: "Sources/Examples/Simple"),
-    .testTarget(name: "SwiftGRPCTests", dependencies: ["SwiftGRPC"]),
-    .testTarget(name: "SwiftGRPCNIOTests", dependencies: ["SwiftGRPC", "SwiftGRPCNIO"])
-  ],
-  swiftLanguageVersions: [.v4, .v4_2, .version("5")],
-  cLanguageStandard: .gnu11,
-  cxxLanguageStandard: .cxx11)
+    name: "SwiftGRPC",
+    products: [
+        .library(name: "SwiftGRPC", targets: ["SwiftGRPC"])
+    ],
+    dependencies: packageDependencies,
+    targets: [
+        .target(name: "SwiftGRPC",
+                dependencies: ["CgRPC", "SwiftProtobuf"]),
+        .target(name: "CgRPC",
+                dependencies: cGRPCDependencies,
+                cSettings: [
+                    .headerSearchPath("../BoringSSL/include", .when(platforms: [.iOS, .macOS, .tvOS, .watchOS])),
+                    .unsafeFlags(["-Wno-module-import-in-extern-c"])],
+                linkerSettings: [.linkedLibrary("z")]),
+        .target(name: "RootsEncoder"),
+        .target(name: "protoc-gen-swiftgrpc",
+                dependencies: [
+                    "SwiftProtobuf",
+                    "SwiftProtobufPluginLibrary",
+                    "protoc-gen-swift"]),
+        .target(name: "BoringSSL"),
+        .target(name: "Echo",
+                dependencies: [
+                    "SwiftGRPC",
+                    "SwiftProtobuf",
+                    "Commander"],
+                path: "Sources/Examples/Echo"),
+        .target(name: "Simple",
+                dependencies: ["SwiftGRPC", "Commander"],
+                path: "Sources/Examples/Simple"),
+        .testTarget(name: "SwiftGRPCTests", dependencies: ["SwiftGRPC"])
+    ],
+    swiftLanguageVersions: [.v4, .v4_2, .version("5")],
+    cLanguageStandard: .gnu11,
+    cxxLanguageStandard: .cxx11)
+
+if isLinux {
+    package.products.append(
+        .library(name: "SwiftGRPCNIO", targets: ["SwiftGRPCNIO"])
```

Comment:
@MrMage what do you think about constraining the NIO version to Linux? Given it's a _really_ old version I don't _think_ it should matter. We could even remove it altogether from `master`?

Reply:
I have also been thinking of just removing the old NIO version from `master` because it causes a bunch of problems. Let's go ahead with that, I guess.

Reply:
@MrMage I suppose tests will have to be removed for linux with swift 5.0 as well? Should it be done in a separate PR?

Reply:
I would suggest removing the entire `SwiftGRPCNIO` directories and targets (for all platforms). I think we could do that in this PR. We could temporarily keep them in a separate branch for those who still need it.

Reply:
@MrMage alright, I've removed all NIO related packages/targets. As a consequence I also removed [linux entry](https://github.com/grpc/grpc-swift/pull/682/files#diff-354f30a63fb0907d4ad57269548329e3L29) from `.travis.yml` file for Swift 5.

Reply:
I don't think you should remove the Linux entry from .travis.yml. But you should remove `Sources/GRPCNIO` and `Tests/GRPCNIOTests` and update the test main file accordingly.

Reply:
@MrMage if `Sources/GRPCNIO` and `Tests/GRPCNIOTest` gets removed, `Package@swift-4.2.swift` won't work, does it mean it has to be removed as well?

Reply:
Why would that not work? Why not just remove the NIO packages from all Package.swift files?

Reply:
The proposed outcome here is to have no `nio`-based version on the master branch at all, so the related targets and dependencies also need to be removed from `Package.swift` and `Package@swift-4.2.swift`.

Reply:
@MrMage what I was referring to was linux build: if you remove all NIO dependencies (including `swift-nio-ssl-support`) from 4.2 then linux won't compile. Do you expect it to compile?

@glbrntt new change-set should contain no nio related dependencies.